packages/realm-react/src/useQuery.tsx

```diff
@@ -29,25 +34,38 @@ import { symbols } from "@realm/common";
  */
 export function createUseQuery(useRealm: () => Realm) {
   return function useQuery<T>(
-    type: string | ({ new (...args: any): T } & Realm.ObjectClass),
-  ): Realm.Results<T & Realm.Object> {
+    type: RealmObjectType<T>, 
+    query: QueryCallback<T> = (collection) => collection, 
+    deps: DependencyList = [],
+  ): RealmResults<T> {
     const realm = useRealm();
 
+    const queryCallback = useMemo(() => {
+      // We want the user of this hook to be able pass in the `query` function inline (without the need to `useCallback` on it)
+      // This means that the query function is unstable and will be a redefined on each render of the component where `useQuery` is used
+      // Therefore we use the `deps` array to memoize the query function internally, and only use the returned `queryCallback`
+      return query;
+    }, deps);
```

Comment:
I think this could be simplified.
```suggestion
      // We want the user of this hook to be able pass in the `query` function inline (without the need to `useCallback` on it)
      // This means that the query function is unstable and will be a redefined on each render of the component where `useQuery` is used
      // Therefore we use the `deps` array to memoize the query function internally, and only use the returned `queryCallback`
    const queryCallback = useCallback(query, deps);
```